### ISSUE TITLE
no-switch-case-fall-through does not recognize block form of switch-case

### ISSUE THREAD

**Issue by MadaraUchiha** [palantir/tslint#2207]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `2.1.5`
- __Running TSLint via__: Visual Studio Code

#### TypeScript code being linted

```ts
  switch (action.type) {

    case actions.ON_AUTH_FORM_FIELD_CHANGE: {
      const {name, value} = action.payload;
      return state.setIn(['form', 'fields', name], value);
    }

    case actions.MAKE_TEMPORARY_USER_SUCCESS: {
      return state.update('ver', (ver: number) => ver + 1);
    }

    // snip
```

with `tslint.json` configuration:

```json
"no-switch-case-fall-through": true
```

#### Actual behavior

TSLint complains about no-switch-case-fall-through, even though the previous case ends with `return`.
Even when I remove the `return` and add a `break`, it won't detect it.

When the curly brackets are removed, both `return` and `break` are recognized, and TSLint stops complaining

#### Expected behavior

TSLint should not complain whether or not the `case:` is followed by an explicit block.

**Comment by ajafff** [comment]
this is a duplicate of #785 
I'm currently rewriting some rules. This one will be the next and include bugfixes for all open bugs. Should be ready this weekend.

**Comment by MadaraUchiha** [comment]
Interesting, I couldn't find it in the search. Thanks for pointing it out! I'm closing this, and will follow the other issue.
